### PR TITLE
[XPU] Support elementwise_min/max/floordiv/where more precision types for KL1 and KL2

### PR DESCRIPTION
**Pull request by newway**
### PR types

New features
### PR changes

OPs
### Description

support elementwise_min/max/floordiv/where op precision types for kunlun2

### CODE REVIEW

**Review by zhupengyang — DISMISSED** [review]
LGTM

File: paddle/phi/kernels/xpu/index_select_kernel.cc [hunk 1]
```diff
@@ -46,8 +46,23 @@ void IndexSelectKernel(const Context& ctx,
...
+  if (index.place() == CPUPlace()) {
+    index_ptr = RAII_GUARD.alloc_l3_or_gm<int8_t>(byte_times * index.numel());
+    PADDLE_ENFORCE_XDNN_NOT_NULL(index_ptr);
+    memory_utils::Copy(ctx.GetPlace(),
+                       reinterpret_cast<void*>(const_cast<int8_t*>(index_ptr)),
+                       CPUPlace(),
+                       reinterpret_cast<const void*>(index.data<int>()),
+                       byte_times * index.numel());
+  } else {
+    index_ptr = index.template data<int8_t>();
```
Comment (hong19860320) [inline paddle/phi/kernels/xpu/index_select_kernel.cc]: 需要根据 index_type 分别用 index.data<int>() 或 index.data<int64_t>()

Comment (newway) [reply paddle/phi/kernels/xpu/index_select_kernel.cc]: done

**Review by hong19860320 — APPROVED** [review]
LGTM

**Pull request merged**